nep-11.mediawiki

```diff
@@ -180,32 +179,13 @@ The returned tokens should be in the following format:
 It must have the fields <code>id</code>, <code>uri</code> and <code>properties</code>.
```

Comment:
This line should be modified too. And can we just return `id`?

Reply:
Oh you're right, I missed that one. Should it not be `id` and `properties`?

Reply:
It's easier for a contract to return an enumerator of `id`.

Reply:
If we return just `id` won't the client then have to get the `properties` in a separate network call? Would it not be more efficient to get the full NFT?

Reply:
Opened a PR @hal0x2328 https://github.com/Splyse/proposals/pull/4

Reply:
Fixed

Reply:
There is currently no way to return an enumerator of a complex data structure. So the best way is just returning the `id`.

Reply:
> There is currently no way to return an enumerator of a complex data structure. So the best way is just returning the `id`.

I've been writing my contracts so that the enumerator returns the objects directly from storage where they are saved as a serialized bytearray, then I use the neon-js `StackItem.deserialize` method to parse them. This way I only need to do a single RPC request instead of dozens or more to get the same data.

Reply:
In this way, the way the object is stored needs to be specified. However, the storage contents of different contracts may not be consistent. So I think the easiest way is to divide it into two steps, the first step returns the enumerator of id, and the second step uses id to get the details.

Reply:
Don't we specify the format in the proposal?

Reply:
Because each contract may have special data structures and storage, our standards should only specify interfaces, and should not specify implementations unless it is necessary.

Reply:
I'm pretty indifferent. @hal0x2328?